plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -410,6 +411,21 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
          }
       }
 
+      void update_block_metrics() {
+         _metrics.unapplied_transactions.value =  _unapplied_transactions.size();
+         _metrics.subjective_bill_account_size.value = _subjective_billing.get_account_cache_size();
+         _metrics.subjective_bill_block_size.value = _subjective_billing.get_block_cache_size();
+         _metrics.blacklisted_transactions.value = _blacklisted_transactions.size();
+         _metrics.unapplied_transactions.value = _unapplied_transactions.size();
+
+         auto& chain = chain_plug->chain();
+         _metrics.last_irreversible.value = chain.last_irreversible_block_num();
+         _metrics.block_num.value = chain.head_block_num();
+
+         const auto& sch_idx = chain.db().get_index<generated_transaction_multi_index,by_delay>();
+         _metrics.scheduled_trxs.value = sch_idx.size();
+      }
```

Comment:
In cases were we have a group like this probably worth while to have an enabled flag that is true only if the the prometheus plugin is enabled.